src/services/data/fetchCategories.js

```diff
@@ -0,0 +1,10 @@
+import axios from "axios";
+
+export const fetchCategories = async ({ videoStateDispatch }) => {
+    try {
+        const res = await axios.get("/api/categories");
+        videoStateDispatch({ type: "ADD_CATEGORIES", payload: res.data.categories });
```

Comment:
Should maintain the checks for various api status code

Reply:
thanks for the suggestion, done